scripts/api_functions.py

```diff
@@ -0,0 +1,91 @@
+import re
+from pathlib import Path
+from typing import List
+
+APIS = ["basic_int", "basic_str", "memview_int", "numpy_int"]
+
+SEPARATOR_TEXT = """\
+###############################
+# Automatically generated API
+# Do not edit below these lines
+###############################
+"""
+
+
+def read_api(api_template_path: Path) -> List[str]:
+    """Read public API from _api_template.py"""
+    with open(api_template_path) as f:
+        lines = f.readlines()
+
+    # Methods within _API_FUNCTIONS class
+    method_lines = [line for line in lines if line.startswith("    def")]
```

Comment:
Can this be done with parsing the file AST? Probably fine to do this way too, especially if we have tests, but AST or reflection would be great.

Reply:
It's definitely possible to do with AST. And actually the easiest approach here might be introspection, since we can run the import. So we should be able to do
```py
from h3.api._api_functions import _API_FUNCTIONS
instance = _API_FUNCTIONS(...)
methods = [
	method for method in dir(instance) if not method.startswith('_')
]
```